imapclient/myrights_test.go

```diff
@@ -0,0 +1,147 @@
+package imapclient_test
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/emersion/go-imap/v2/imapclient"
+	"github.com/emersion/go-imap/v2/internal/imapwire"
+)
+
+func TestMyrights(t *testing.T) {
+	type testcaseType struct {
+		Mailbox string
+		Rights  string
+	}
+
+	clientConn, serverConn := net.Pipe()
+	client := imapclient.New(clientConn, nil)
+
+	defer client.Close()
+	defer serverConn.Close()
+
+	// for server
+	serverErrChan := make(chan error)
+	testcaseChan := make(chan testcaseType)
+
+	defer close(testcaseChan)
+
+	// for client
+	clientErrChan := make(chan error)
+
+	// server
+	go func() {
```

Comment:
Hmmm… I'm not a fan of the ad-hoc server code, it would be nicer to use our own server impl instead…

Reply:
I've added support on the server side for MYRIGHTS and SETACL. But right now SETACL is not checking real meaning of rights on a mailbox, just setting, what's told. I hope its fine(

Reply:
Also note that the ACL extension isn't exposed by default by Dovecot, so some slight configuration changes would be required to use that for ACL testing: https://doc.dovecot.org/configuration_manual/acl/#acl